file/file.go

```diff
@@ -51,38 +46,28 @@ type MetricToPublish struct {
 	Data      interface{}       `json:"data"`
 	Unit      string            `json:"unit"`
 	Tags      map[string]string `json:"tags"`
-	Version_  int               `json:"version"`
-	// Last advertised time is the last time the snap agent was told about a metric.
-	LastAdvertisedTime time.Time `json:"last_advertised_time"`
+	Version   int64             `json:"version"`
 }
 
-//NewFilePublisher returns an instance of filePublisher
-func NewFilePublisher() *filePublisher {
+//New returns an instance of filePublisher
+func New() *filePublisher {
 	return &filePublisher{}
 }
 
-func (f *filePublisher) Publish(contentType string, content []byte, config map[string]ctypes.ConfigValue) error {
+func (f *filePublisher) Publish(mts []plugin.Metric, cfg plugin.Config) error {
 	logger := log.New()
-	logger.Println("Publishing started")
-	var mts []plugin.MetricType
-
-	switch contentType {
-	case plugin.SnapGOBContentType:
-		dec := gob.NewDecoder(bytes.NewBuffer(content))
-		if err := dec.Decode(&mts); err != nil {
-			logger.Printf("Error decoding: error=%v content=%v", err, content)
-			return fmt.Errorf("Error decoding %v", err)
-		}
-	default:
-		return fmt.Errorf("Unknown content type '%s'", contentType)
+	logger.Debug("Publishing started")
+
+	destination, err := cfg.GetString("file")
+	if err != nil {
+		return fmt.Errorf("%s: %s", err, "file")
 	}
 
-	logger.Printf("publishing %v metrics to %v", len(mts), config)
-	file, err := os.OpenFile(config["file"].(ctypes.ConfigValueStr).Value, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0666)
+	logger.Infof("Publishing %v metrics to %s", len(mts), destination)
```

Comment:
The previous log is with ``Debug` level, this with `Info` level. Have you set different log level intentionally?